### PR TITLE
Users/ersciple/m110normalizefind

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: node/mock-task.ts [hunk 1]
```diff
@@ -74,8 +74,13 @@ export function setResourcePath(path: string): void {
     // nothing in mock
 }
 
-export function loc(key: string): string {
-    return 'loc_mock_' + key;
+export function loc(key: string, ...args: any[]): string {
+    let str: string = 'loc_mock_' + key;
+    if (args.length) {
```
Comment (TingluoHuang) [inline node/mock-task.ts]: try args is empty.

File: node/mock-task.ts [hunk 2]
```diff
@@ -397,21 +368,33 @@ export interface MatchOptions {
     nonegate?: boolean;
     flipNegate?: boolean;
 }
-export function match(list: string[], pattern: string, options?: MatchOptions): string[];
-export function match(list: string[], patterns: string[], options?: MatchOptions): string[];
-export function match(list: string[], pattern: any, options?: MatchOptions): string[] {
-    return mock.getResponse('match', pattern) || [];
+
+export function match(list: string[], patterns: string[] | string, patternRoot?: string, options?: MatchOptions): string[] {
+    let responseKey: string = typeof patterns == 'object' ? (patterns as string[]).join('\n') : patterns as string;
```
Comment (TingluoHuang) [inline node/mock-task.ts]: try pass in multiple patterns.

File: node/mock-task.ts [hunk 3]
```diff
@@ -397,21 +368,33 @@ export interface MatchOptions {
...
+    let result: string[] = mock.getResponse('match', responseKey, module.exports.debug);
```
Comment (TingluoHuang) [inline node/mock-task.ts]: maybe not mock at all, since it's not touch disk, and user has full control of all inputs.

Comment (ericsciple) [inline node/mock-task.ts]: sounds good

**Pull request merged**